src/core.js

```diff
@@ -1337,7 +1337,7 @@ define([
         var _this = this;
         mug.on("messages-changed", function (event) {
             _this.setTreeValidationIcon(event.mug);
-        }, null, this.data.core);
+        }, null, null, this.data.core);
```

Comment:
This is the ugly one, but this is also the only place that this variant of the API is used.